### ISSUE TITLE
Terrain that renders a resource sould have a finite amount of said resource

### ISSUE THREAD

**Issue by dusktreader** [zudijyr/Building-Sim-Game#13]
For example, a tile of forest should have a fixed amount of wood that is depleted as a unit chops wood on it.

Maybe this resource can be regenerated slowly over time, but resources should be able to be depleted.

**Comment by zudijyr** [comment]
Agreed.  Although we should have a global setting for how quickly they get depleted, so people can have infinite resources if they don't want to deal with that.

**Comment by dusktreader** [comment]
Yeah, maybe just a global flag. So, if it is set to unlimited, the code checks for the flag before computing depletion
